databroker/core.py

```diff
@@ -833,3 +833,90 @@ def __iter__(self):
                 yield doc
             else:
                 self.__stash_values(name, doc)
+
+
+class HeaderSourceShim(object):
```

Comment:
This is not how the mds API works. Instead you need `find_run_start`, `find_run_stop`, etc. The slicing and __call__ logic are implemented once in Broker, not per variant of MDS.